MoodDetectorWebApp/Global.asax.cs

```diff
@@ -12,7 +17,39 @@ protected void Application_Start()
             AreaRegistration.RegisterAllAreas();
             FilterConfig.RegisterGlobalFilters(GlobalFilters.Filters);
             RouteConfig.RegisterRoutes(RouteTable.Routes);
-            BundleConfig.RegisterBundles(BundleTable.Bundles);
+            BundleConfig.RegisterBundles(BundleTable.Bundles);         
         }
+
+        protected void Application_PostAuthenticateRequest(Object sender, EventArgs e)
+        {
+            if (FormsAuthentication.CookiesSupported == true)
+            {
+                if (Request.Cookies[FormsAuthentication.FormsCookieName] != null)
+                {
+                    try
+                    {                                     
+                        string username = FormsAuthentication.Decrypt(Request.Cookies[FormsAuthentication.FormsCookieName].Value).Name;
+                        string roles = string.Empty;
+
+                        using (MoodDetectorDBEntities entities = new MoodDetectorDBEntities())
+                        {
+                            LoginInfo user = entities.LoginInfoes.SingleOrDefault(u => u.Username == username);
+
+                            roles = user.User.AccessRights;
+                        }
+                  
+                        HttpContext.Current.User = new System.Security.Principal.GenericPrincipal(
+                        new System.Security.Principal.GenericIdentity(username, "Forms"), roles.Split(';'));
+                        
+                    }
+                    catch (Exception)
+                    {
+                       
```

Comment:
I don't think you should leave it like that. If you manage possible exceptions properly, we could check one more requirement from the list.